os/board/rtl8730e/src/component/soc/amebad2/fwlib/ap_core/ameba_app_start.c

```diff
@@ -21,6 +20,34 @@
 
 #include "ameba_soc.h"
 
+#ifdef CONFIG_APP_BINARY_SEPARATION
+/* In case of app binary separation, we need separate page tables for each app.
+ * So, the page table size is set here based on the following conditions:
+ * 1. One L1 page table (16KB) for kernel
+ * 2. One L1 page table for each app
+ * 3. L2 page tables for each app and common binary
+ * 4. The number of L2 page tables required for the app depends on the size of the 
+ * apps. For example, if the app is 2MB, then it requires minimum of 2 L2 page tables
+ * (since each L2 table can map 1 MB memory). However, we are currently reserving 
+ * space assuming each app and common binary require 4 L2 page tables
+ * 5. The start address of first kernel L1 page table must be 16KB aligned
+ *
+ * Based on the above considerations, we setup the page table size as 
+ * 1 Kernel L1 table + 2 app L1 tables + (2 apps + 1 common) * 4 L2 page tables
+ * 16 KB + 16 KB + 16 KB + 3 * 4 * 1 KB = 60 KB
+ * 60 KB = 0xF000 ==> Round it up to 16KB boundary ==> 0x10000
+ */
+#define PGTABLE_SIZE       0x00010000
```

Comment:
How about using the newly added config CONFIG_NUM_L2_PER_APP here so that user need not change this whenever the config is changed?

Reply:
I tried this, but this is causing several other issues. 
Even if we make this change, there are many other things like page size, etc which are defined in other header files. 
The main problem is that these values are defined in mmu.h and used in other assembly files of the target chipset. So, any attempt to use complex definition directives is resulting in multiple errors. So let us keep the current status.